src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,379 @@
+            kernel_size=config.head_pooling_size, stride=1, padding=(config.head_pooling_size - 1) // 2
+        )
+        self.pooling_2s = nn.MaxPool2d(
+            kernel_size=config.head_pooling_size // 2 + 1, stride=1, padding=(config.head_pooling_size // 2) // 2
+        )
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(FAST_FOR_CAPTIONING_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=FastForSceneTextRecognitionOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: torch.FloatTensor,
+        output_hidden_states: Optional[bool] = True,
+        return_dict: Optional[bool] = None,
+        labels: Optional[Dict] = None,
+    ):
+        r"""
+        labels (`Dict[str, torch.Tensor]`, *optional*):
+            Should contain 3 keys: gt_texts, gt_kernels, gt_instances
+
+        Returns:
+
+                Examples:
+
+        ```python
+        >>> from transformers import FastImageProcessor, FastForSceneTextRecognition
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> url = "https://huggingface.co/datasets/Raghavan/fast_model_samples/resolve/main/img657.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw).convert("RGB")
+        >>> processor = FastImageProcessor.from_pretrained("jadechoghari/fast-tiny")
+        >>> model = FastForSceneTextRecognition.from_pretrained("jadechoghari/fast-tiny")
+        >>> inputs = processor(image, return_tensors="pt")
+        >>> # forward pass
+        >>> outputs = model(pixel_values=inputs["pixel_values"])
+        >>> target_sizes = [(image.height, image.width)]
+        >>> threshold = 0.88
+        >>> text_locations = processor.post_process_text_detection(outputs, target_sizes, threshold, bounding_box_type="rect")
+        >>> print(text_locations[0]["boxes"][0])
+        [151, 151, 160, 56, 355, 74, 346, 169]
+        ```
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        features = (
+            self.backbone(pixel_values) if self.config.use_timm_backbone else self.backbone(pixel_values).feature_maps
+        )
+
+        hidden_states = self.neck(features)
+
+        text_detection_output = self.text_detection_head(hidden_states)
+
+        all_hidden_states = (features, hidden_states)
+
+        loss = None
+        if labels:
+            upsampled_output = F.interpolate(
+                text_detection_output, size=(pixel_values.size(2), pixel_values.size(3)), mode="bilinear"
+            )
+            loss = self.loss_function()
+            kernels = upsampled_output[:, 0, :, :]  # 4*640*640
+            texts = self.pooling_1s(kernels)  # 4*640*640
+            loss = loss(upsampled_output, labels, texts)
+
+        text_detection_output = F.interpolate(
+            text_detection_output, size=(pixel_values.size(2) // 4, pixel_values.size(3) // 4), mode="bilinear"
+        )
```

Comment:
Why not to interpolate here for the `pixel_values` size and avoid later interpolation in post processing